src/helpers.ts

```diff
@@ -1,5 +1,44 @@
+// Courtesy: https://stackoverflow.com/a/23945027
+function extractHostname(url) {
+  var hostname;
+  //find & remove protocol (http, ftp, etc.) and get hostname
+
+  if (url.indexOf("//") > -1) {
+    hostname = url.split('/')[2];
+  } else {
+    hostname = url.split('/')[0];
+  }
+
+  //find & remove port number
+  hostname = hostname.split(':')[0];
+  //find & remove "?"
+  hostname = hostname.split('?')[0];
+
+  return hostname;
+}
+
+// Warning: you can use this function to extract the "root" domain, but it will not be as accurate as using the psl package. 
+// https://www.npmjs.com/package/psl
+const  extractRootDomain = (url) => {
+  var domain = extractHostname(url),
+  splitArr = domain.split('.'),
+  arrLen = splitArr.length;
+
+  //extracting the root domain here
+  //if there is a subdomain
+  if (arrLen > 2) {
+    domain = splitArr[arrLen - 2] + '.' + splitArr[arrLen - 1];
+    //check to see if it's using a Country Code Top Level Domain (ccTLD) (i.e. ".me.uk")
+    if (splitArr[arrLen - 2].length == 2 && splitArr[arrLen - 1].length == 2) {
+      //this is using a ccTLD
+      domain = splitArr[arrLen - 3] + '.' + domain;
+    }
+  }
+  return domain;
+}
 export const getCookieDomain = () => {
-  return location.hostname.replace('www.', '');
+  return `.${extractRootDomain(location.hostname)}` // .localhost
```

Comment:
For the cookie domain, if the user has disabled it, we have to check here?

Reply:
No, the parent check is for cross_subdomain_cookie. The method loses its significance if cross_subdomain_cookie is set to false.